### DISCUSSION TITLE
What is the best practice solution for implementing SvelteKit with SSR in Supabase?

### DISCUSSION THREAD

**Discussion by samhsabin** [supabase/supabase#20603] [Questions]
I want to use Sveltekit with Supabase using SSR, but I'm confused by the documentation.

Most of the example use Auth helpers(which it also explains in the documentation is not recommended, saying you should use the new @supabase/ssr instead).

For the documentation that does address SSR, there is a warning message saying you should use auth.getUser instead of accessing the session object for server side rendering, and then proceeds to show you how to set things up using the session object despite explaining that doing it that way can cause a security issue.

Is there no documentation showing how to set things up correctly??

From what I've seen from external documentation, it appears that best practice is to set a hook system that checks auth.getUser before any requests to Supabase. Is that correct, and if so, how would it be done?

**Comment by CropWatchDevelopment** [comment]
I am also looking for an answer to this, their examples are missing Sveltekit examples :(
